File: C++/largest_3_same_digit_number_in_string.cpp
// no.2264
#include <string>
using namespace std;

class Solution {
public:
    string largestGoodInteger(string num) {
        char can = num[0];
        char ans = ' ';
        int count = 1;
        for(int i = 1; i < num.size(); i++){
            if(num[i] == can){
                count ++;
            }else{
               count = 1;
               can = num[i]; 
            }
            if(count == 3 & ans < can){
                ans = can;
            }
            
        }
        return ans == ' ' ? "" : string(3, ans);
    }
};

File: python/largest_3_same_digit_number_in_string.py
# no. 2264
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        can = num[0]
        counter = 1
        ans = ''
        for i in num[1:]:
            if can == i:
                counter += 1
            else:
                can = i
                counter = 1
            if counter == 3 and ans < can:
                ans = can
        return ans * 3 if ans else ans
